test/datasets_utils.py

```diff
@@ -313,7 +313,7 @@ def create_dataset(
 
         special_kwargs, other_kwargs = self._split_kwargs(kwargs)
         if "download" in self._HAS_SPECIAL_KWARG:
-            special_kwargs["download"] = False
+            special_kwargs["download"] = None if self.DATASET_CLASS.__name__ == 'ImageNet' else False
```

Comment:
Why this change? With `download=False` the dataset will emit a warning, but should not behave anything different. Given that this warning is emitted for a long time now, I wonder if we can remove the `download` flag from `ImageNet` all together.

If we really want to avoid, I suggest we change L316 to honor a explicitly passed `download`

```python
special_kwargs.setdefault("download", False)
```
and overwrite `create_dataset` in the ImageNetTestCase to always pass `download=None`

```python
@contextlib.contextmanager
def create_dataset(self, *args, **kwargs):
    kwargs.setdefault("download", None)
    with super().create_dataset(*args, **kwargs) as (dataset, info):
        yield dataset, info
```

Reply:
> Why this change?

It's to avoid the warning

> Given that this warning is emitted for a long time now, I wonder if we can remove the download flag from ImageNet all together.

Agreed. Thoughs @fmassa? The warning has been introduced in https://github.com/pytorch/vision/pull/1457 in v0.5

Reply:
> and overwrite create_dataset in the ImageNetTestCase to always pass download=None

I went for a simpler solution which is to only override the default if `download` exists and if its default is True-y. This way, when the default is False or None, it doesn't get overridden.

This still avoids hardcoding the 'ImageNet' name which I believe was the main issue here